core/shared/src/main/scala/scalaz/zio/stm/STM.scala

```diff
@@ -575,6 +571,8 @@ object STM {
       protected[this] val expected: Versioned[A]
       protected[this] var newValue: A
 
+      val isNew: Boolean
```

Comment:
Nice !

Reply:
😄 

If TRef is created inside the transaction, then it will not exist until transaction committed, so we don't need to add suspended transaction to `todo` list of new TRef. Now the `diffJournal` method just ignores all the "new" TRefs created in that transaction so process will terminate even without stable identifiers.

Also I deleted all TRef identities, since we can just use the `TRef` itself as the identity in the hash map. It cleaned up code and improved performance. 💃